CMS-Enterprise/easi-app
# [EASI-4639] Discussion views cleanup

# EASI-4639

## Description
- Fixed bug where discussions were not refetched after form submit
  - Added new `GetSystemIntakeGRBDiscussions.ts` query
- Added `MentionTextArea` to `DiscussionPost`
  - Truncated text with read more/less button
- `MentionTextArea` fixes:
  - Fixed ref prop - passed form ref to component and added `tabIndex` attr to allow for focusing on div
  - Removed text area styling from component when `editable` is set to `false`
  - Fix bug where field text does not clear when form is submitted
- Unit tests

## How to test this change
- Test using intake titled "System Intake with some different GRB Reviewers (and discussions)" from seeded data
- Go to Discussions section of GRB review tab
- Latest discussion content should match [Figma](https://www.figma.com/design/G0UN35V7SY0g9nkTIESxsY/EASi-GRB-Experience?node-id=3275-14255&t=D02GhLBVjEQFp4kl-1)
- Click "View discussion board" to open panel to reply view
- Fill out and submit reply form
- Discussions should refetch and show updated replies
- Text should clear from field on form submit

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
